src/pages/experiments/[experimentId]/data-processing/configure-embedding/components/CalculationConfig.jsx

```diff
@@ -39,20 +42,25 @@ const CalculationConfig = (props) => {
   }, 1500), []);
 
   const updateSettings = (diff) => {
-    // Always update the settings in the Redux store.
-    dispatch(updateProcessingSettings(
-      experimentId,
-      FILTER_UUID,
-      diff,
-    ));
```

Comment:
it looks like this is done in both cases, why is it refactored and put under each of the if statements?

Reply:
Correct. The reason is that in the former case, the dispatch is executed as a callback, notice that it is embedded as the second argument of `setChangesOutstanding`.

When you use `useState()`, there is no guarantee that the setter will be called synchronously with the rest of the code, i.e. it is possible for `changesOutstanding` to be `false` by the time the update Redux action is called. This caused some timing issues, e.g. it would think that the user pressed `Apply` and therefore the embedding could be reloaded when that was not the case.

The way it's resolved is to put the dispatch into a callback, which is provided as the second argument, e.g.:

```
setValue(false, () => console.log('ran after value set to false'))
```

Reply:
thanks for the clarification, makes sense now